File: zig/src/apu.zig
const SDL = @import("sdl2");

pub const APU = struct {
    silent: bool,
    debug: bool,

    pub fn new(silent: bool, debug: bool) !APU {
        if (!silent) {
            try SDL.init(.{ .audio = true });
        }
        return APU{
            // FIXME
            .silent = silent,
            .debug = debug,
        };
    }
    pub fn tick(self: APU) !void {
        _ = self;
    }
};

File: js/src/apu.ts
import { RAM } from "./ram";

export class APU {
    ram: RAM;
    silent: boolean;
    debug: boolean;

    constructor(ram: RAM, silent: boolean, debug: boolean) {
        this.ram = ram;
        this.silent = silent;
        this.debug = debug;
    }

    tick() {}
}
